tools/toolchain/scripts/install_cosma.sh

```diff
@@ -0,0 +1,155 @@
+                [ -f libs/grid2grid/src/grid2grid/*.a ] && install -m 644 libs/grid2grid/src/grid2grid/*.a ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+                [ -f libs/options/*.so ] && install -m 644 libs/options/*.so ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+                install -m 644 src/cosma/*.a ${pkg_install_dir}/lib/cuda >> install.log 2>&1
+            fi
+        fi
+        COSMA_ROOT="${pkg_install_dir}"
+        COSMA_CFLAGS="-I'${pkg_install_dir}/include'"
+
+        # check if cosma is compiled with 64bits and set up COSMA_LIBDIR accordingly
+        COSMA_LIBDIR="${pkg_install_dir}/lib"
+
+        [ -d  ${pkg_install_dir}/lib64 ] && COSMA_LIBDIR="${pkg_install_dir}/lib64"
+
+        COSMA_LDFLAGS="-L'${COSMA_LIBDIR}' -Wl,-rpath='${COSMA_LIBDIR}'"
+        COSMA_CUDA_LDFLAGS="-L'${COSMA_LIBDIR}/cuda' -Wl,-rpath='${COSMA_LIBDIR}/cuda'"
+        ;;
+    __SYSTEM__)
+        echo "==================== Finding cosma from system paths ===================="
+        check_command pkg-config --modversion cosma
+        add_include_from_paths COSMA_CFLAGS "cosma.h" $INCLUDE_PATHS
+        add_lib_from_paths COSMA_LDFLAGS "libcosma.*" $LIB_PATHS
+        ;;
+    __DONTUSE__)
+        ;;
+    *)
+        echo "==================== Linking cosma to user paths ===================="
+        pkg_install_dir="$with_cosma"
+        check_dir "$pkg_install_dir/lib"
+        check_dir "$pkg_install_dir/lib64"
+        check_dir "$pkg_install_dir/include"
+
+
+        # check if cosma is compiled with 64bits and set up COSMA_LIBDIR accordingly
+        COSMA_LIBDIR="${pkg_install_dir}/lib"
+
+        [ -d  ${pkg_install_dir}/lib64 ] && COSMA_LIBDIR="${pkg_install_dir}/lib64"
+
+        COSMA_CFLAGS="-I'${pkg_install_dir}/include'"
+        COSMA_LDFLAGS="-L'${COSMA_LIBDIR}' -Wl,-rpath='${COSMA_LIBDIR}'"
+        ;;
+esac
+if [ "$with_cosma" != "__DONTUSE__" ] ; then
+    COSMA_LIBS="-lcosma_with_scalapack -lgrid2grid -lcosma -loptions"
+    if [ "$with_cosma" != "__SYSTEM__" ] ; then
+        cat << EOF > "${BUILDDIR}/setup_cosma"
+prepend_path LD_LIBRARY_PATH "${COSMA_LIBDIR}"
+prepend_path LD_RUN_PATH "${COSMA_LIBDIR}"
+prepend_path LIBRARY_PATH "${COSMA_LIBDIR}"
+prepend_path CPATH "$pkg_install_dir/include"
+export COSMA_INCLUDE_DIR="$pkg_install_dir/include"
+export COSMA_LIBS="-lcosma_with_scalapack -lgrid2grid -lcosma -loptions"
+export COSMA_ROOT="${pkg_install_dir}"
+export PKG_CONFIG_PATH="$PKG_CONFIG_PATH:${COSMA_LIBDIR}/pkgconfig"
+EOF
+    fi
+    cat << EOF >> "${BUILDDIR}/setup_cosma"
+export COSMA_CFLAGS="${COSMA_CFLAGS}"
+export COSMA_LDFLAGS="${COSMA_LDFLAGS}"
+export COSMA_CUDA_LDFLAGS="${COSMA_CUDA_LDFLAGS}"
+export CP_DFLAGS="\${CP_DFLAGS} IF_MPI(IF_OMP(-D__COSMA|)|)"
+export CP_CFLAGS="\${CP_CFLAGS} ${COSMA_CFLAGS}"
+export CP_LDFLAGS="\${CP_LDFLAGS} IF_CUDA(${COSMA_CUDA_LDFLAGS}|${COSMA_LDFLAGS})"
+export COSMA_LIBRARY="-lcosma_with_scalapack -lgrid2grid -lcosma -loptions"
```

Comment:
this one should be: 
`-lcosma_pxgemm -lcosma -lgrid2grid` for the CPU-only version and
-lcosma_pxgemm -lcosma -lgrid2grid -lTiled-MM` for the GPU-accelerate version